src/etc/inc/services.inc

```diff
@@ -249,51 +249,53 @@ function services_radvd_configure($blacklist = array()) {
 		$radvdconf .= "\t};\n";
 
 		/* add DNS servers */
-		$dnslist = array();
-		if (isset($dhcpv6ifconf['rasamednsasdhcp6']) && is_array($dhcpv6ifconf['dnsserver']) && !empty($dhcpv6ifconf['dnsserver'])) {
-			foreach ($dhcpv6ifconf['dnsserver'] as $server) {
-				if (is_ipaddrv6($server)) {
-					$dnslist[] = $server;
+		if (($dhcpv6ifconf['radvd-dns']) != 'disabled') {
```

Comment:
One last little nit I didn't catch before, the variable doesn't need to be surrounded by ``()`` on its own like that.